python/metrics/src/hydrotools/metrics/metrics.py

```diff
@@ -112,6 +113,62 @@ def nash_sutcliffe_efficiency(
         return 1.0 / (1.0 + numerator/denominator)
     return 1.0 - numerator/denominator
 
+def kling_gupta_efficiency(
+    y_true: npt.ArrayLike,
+    y_pred: npt.ArrayLike,
+    r_scale: float = 1.0,
+    a_scale: float = 1.0,
+    b_scale: float = 1.0
+    ) -> float:
+    """Compute the Kling-Gupta model efficiency coefficient (KGE).
```

Comment:
Should any software assumptions of be stated here? Something like: "This function assumes two sorted arrays of equal length." I would also want to add something that conveys that `y_true[0]` "lines up with" `y_pred[0]`. I cant find an appropriate phrase though.

Reply:
This made this realize, the docstring description for these doesn't make any sense. I copied these from `sklearn` and forgot to update them for this case.

Reply:
Updated docstrings and added new input array validation methods for NSE and KGE. These could certainly be way more rigorous and exhaustive, but it's a start.

Reply:
Note: the parameters descriptions explicitly state the expected shape (`(n_samples,)`).